doc/ufunc_examples.rst

```diff
@@ -0,0 +1,192 @@
+.. _ufunc_examples:
+
+Ufunc Examples
+--------------
+
+This page contains examples of different Grid Ufuncs that you might find useful.
+They are intended to be more advanced or realistic cases than the simple differencing operations we showed in the introduction to :ref:`grid_ufuncs`.
+
+As each of these cases is a vector calculus operator, we will demonstrate their use on an example two-dimensional vector field.
+
+Firstly we need a two-dimensional grid, and we use similar coordinate names to the 1D example we used when first introducing :ref:`grids`.
+
+.. ipython:: python
+
+    import numpy as np
+    import xarray as xr
+
+    grid_ds = xr.Dataset(
+        coords={
+            "x_c": (
+                [
+                    "x_c",
+                ],
+                np.arange(1, 20),
+                {"axis": "X"},
+            ),
+            "x_g": (
+                [
+                    "x_g",
+                ],
+                np.arange(0.5, 19),
+                {"axis": "X", "c_grid_axis_shift": -0.5},
+            ),
+            "y_c": (
+                [
+                    "y_c",
+                ],
+                np.arange(1, 20),
+                {"axis": "Y"},
+            ),
+            "y_g": (
+                [
+                    "y_g",
+                ],
+                np.arange(0.5, 19),
+                {"axis": "Y", "c_grid_axis_shift": -0.5},
+            ),
+        }
+    )
+    grid_ds
```

Comment:
I feel l like there must be a more compact way to write this. Perhaps not using black formatting would be better here.

Reply:
I think removing the trailing commas behind e.e. `y_g` makes this a lot nicer.

Reply:
I actually think we should consider turn black formatting off across our docs in general. 1) it causes problems interacting with ipython-sphinx (it adds blank lines that ipython interprets as the end of a code block),
2) one intention of black is to put every argument on a different line, so that `git diffs` are as clear as possible. But you don't really need that for docs examples, and human readability is more important than the clarity of a diff in that case.